### ISSUE TITLE
Empty GREP_OPTIONS for grep alias

### ISSUE THREAD

**Issue by amp1ee** [ohmybash/oh-my-bash#120]
Observed that `grep` loses its `--color=auto` option:
```
$ alias grep
alias grep='grep '
```
Added `set -x` flag to `${OSH}/lib/grep.sh` to see what's going on:
```
$ src
++++ GREP_OPTIONS=
++++ grep_flag_available --color=auto
++++ grep --color=auto ''
++++ echo
++++ GREP_OPTIONS+=(" --color=auto")
++++ VCS_FOLDERS='{.bzr,CVS,.git,.hg,.svn}'
++++ grep_flag_available --exclude-dir=.cvs
++++ grep --exclude-dir=.cvs ''
++++ echo
++++ GREP_OPTIONS+=(" --exclude-dir=$VCS_FOLDERS")
++++ alias 'grep=grep '
++++ unset GREP_OPTIONS
++++ unset VCS_FOLDERS
++++ unset -f grep_flag_available
++++ set +x
```
As can be seen from the output, alias for grep is just `'grep '` in result.
The problem is that the GREP_OPTIONS is initialized like a string variable and then used like an array. I suggest to declare this variable as an array in the first place:
```declare -a GREP_OPTIONS```

My bash version is 4.3.48(1)-release.
CC: @nntoan

**Comment by allan4229** [comment]
Confirmed, it affects me too. I've just found this issue after getting mad at my grep colors not working properly.

For me , the solution was to just remove the parentheses of the right value.

So, this (File ```${OSH}/lib/grep.sh```) :
```sh
GREP_OPTIONS+=(" --color=auto")
```
Turns into this :
```sh
GREP_OPTIONS+=" --color=auto"
```

Or simply :
```sh
perl -pe 's/([\(]\s*\")([^"]+)(\"\s*[\)])/"$2"/' -i "${OSH}/lib/grep.sh"
```

**Comment by akinomyoga** [comment]
This has been fixed by #170.
